app/scripts/controllers/recent-blocks.js

```diff
@@ -39,6 +37,73 @@ class RecentBlocksController {
 
     this.store.updateState(state)
   }
+
+  backfillBlock (newBlock) {
+    const block = this.mapTransactionsToPrices(newBlock)
+
+    const state = this.store.getState()
+
+    if (state.recentBlocks.length < this.historyLength) {
+      state.recentBlocks.unshift(block)
+    }
+
+    this.store.updateState(state)
+  }
+
+  mapTransactionsToPrices (newBlock) {
+    const block = extend(newBlock, {
+      gasPrices: newBlock.transactions.map((tx) => {
+        return tx.gasPrice
+      }),
+    })
+    delete block.transactions
+    return block
+  }
+
+  async backfill() {
+    this.blockTracker.once('block', async (block) => {
+      let blockNum = block.number
+      let recentBlocks
+      let state = this.store.getState()
+      recentBlocks = state.recentBlocks
+
+      while (recentBlocks.length < this.historyLength) {
+        try {
+          let blockNumBn = new BN(blockNum.substr(2), 16)
+          const newNum = blockNumBn.subn(1).toString(10)
+          const newBlock = await this.getBlockByNumber(newNum)
```

Comment:
the way this is setup, we're backfilling in series, but we can backfill in parallel if we want

Reply:
I did this to make the spike gentler on infura. Maybe misled. Even added 100ms pause between blocks.